keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1108 @@
+If any of the resources can be exhausted, how this is mitigated with the existing limits
+(e.g. pods per node) or new limits added by this KEP?
+
+Are there any tests that were run/should be run to understand performance characteristics better
+and validate the declared limits?
+-->
+Yes. With pod-level resource limits, it's possible to pack more pods onto a node. This increased
+pod density can also lead to higher consumption of resources like PIDs and sockets, as each
+pod and its containers create processes and potentially establish network resources.
+
+
+### Troubleshooting
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
+-->
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+No dependency on etcd availability.
+
+The scheduler relies on the API server to fetch pod specifications and node information.
+Without access to the API server, it cannot make informed scheduling decisions based on
+pod-level resource requests. This could lead to suboptimal placement.
+
+###### What are other known failure modes?
+
+<!--
+For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+-->
+
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+It is suggested that if workloads with pod level resource specifications enabled cause performance 
+or stability degradations, those workloads are recrated with only container level resource specs.
+
+## Implementation History
+
+- ** 2020-03-17:** Initial discussion in [SIG Node meeting] (https://www.youtube.com/watch?v=3cU56ZiUZ8w&list=PL69nYSiGNLP1wJPj5DYWXjiArF-MJ5fNG&index=101)
+- ** 2020-03-30:** Initial [KEP draft](https://github.com/kubernetes/enhancements/pull/1592) and discussion.
+- ** 2021-07-26:** Issue [link](https://github.com/kubernetes/enhancements/issues/2837)
+- ** 2024-05-31:** Revised KEP for alpha (#4678)[https://github.com/kubernetes/enhancements/pull/4678]
+
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+## Alternatives
```

Comment:
there is no really a good alternative for this, are there? Please tell so here

Reply:
For simplifying the user experience, pod resizing could be one alternative. But you need to integrate that with autoscaling, and there's still additional delay that's needed to adjust the resources. There's also the risk of the resizing request being denied.